platforms/documentation/docs/src/docs/userguide/platforms/jvm/toolchains.adoc

```diff
@@ -33,101 +33,214 @@ You can also control the JVM used to run Gradle itself using the <<gradle_daemon
+The `java` block is flexible and supports additional configuration options.
+You can learn more in <<sec:using-java-toolchains>>.
 
-1. Gradle configures all compile, test and javadoc tasks to use the defined toolchain.
-2. Gradle detects <<#sec:auto_detection,locally installed toolchains>>.
-3. Gradle chooses a toolchain matching the requirements (any Java 17 toolchain for the example above).
-4. If no matching toolchain is found, Gradle can automatically download a matching one based on the configured <<#sub:download_repositories,toolchain download repositories>>.
+[[sec:flag-toolchain]]
+=== 2. The `--release` flag
 
-[NOTE]
-====
-Toolchain support is available in the Java plugins and for the tasks they define.
+For strict cross-compilation, the `--release` flag is recommended instead of `sourceCompatibility` and `targetCompatibility`:
 
-For the Groovy plugin, compilation is supported but not yet Groovydoc generation.
-For the Scala plugin, compilation and Scaladoc generation are supported.
+====
+[.multi-language-sample]
+=====
+[source,kotlin]
+----
+tasks.withType<JavaCompile>().configureEach {
+    options.release = 8
+}
+----
+=====
+[.multi-language-sample]
+=====
+[source,groovy]
+----
+tasks.withType(JavaCompile).configureEach {
+    options.release = 8
+}
+----
+=====
 ====
 
-[[sec:vendors]]
-=== Selecting toolchains by vendor
+This flag prevents accidental use of newer APIs that are not available in the specified version.
+However, it does not control which JDK is used—only how the compiler treats source code.
 
-In case your build has specific requirements from the used JRE/JDK, you may want to define the vendor for the toolchain as well.
-link:{javadocPath}/org/gradle/jvm/toolchain/JvmVendorSpec.html[`JvmVendorSpec`] has a list of well-known JVM vendors recognized by Gradle.
-The advantage is that Gradle can handle any inconsistencies across JDK versions in how exactly the JVM encodes the vendor information.
+This method can be combined with toolchains **if you need both a specific JDK and strict cross-compilation**.
+
+[[sec:source-target-toolchain]]
+=== 3. Source and Target compatibility
+
+Setting `sourceCompatibility` and `targetCompatibility` tells the Java compiler to produce bytecode compatible with a specific Java version but does *not* enforce which JDK Gradle itself runs with:
 
 ====
-include::sample[dir="snippets/java/toolchain-filters/kotlin/",files="build.gradle.kts[tags=toolchain-known-vendor]"]
-include::sample[dir="snippets/java/toolchain-filters/groovy/",files="build.gradle[tags=toolchain-known-vendor]"]
+[.multi-language-sample]
+=====
+[source,kotlin]
+----
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+----
+=====
+[.multi-language-sample]
+=====
+[source,groovy]
+----
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
```

Comment:
💭  Does happen often that Kotlin and Groovy snippets are identical? Would it make sense to add a feature to our multi-language-sample thing to display the sample in both languages, but specify it only once in the source?